### ISSUE TITLE
Redundant dependencies

### ISSUE THREAD

**Issue by dsteindo** [mderevyankoaqa/jmeter-influxdb2-listener-plugin#6]
Hello,

first of all I have to say this plugin is amazing, and the Grafana integration is working smoothly.

My current environment setup
- JMeter Version: 5.4.1
- Operation System: Windows 10
- Java Version: sapmachine-jdk-11.0.5

But there is an issue when I try following in JMeter GUI: 
Right click on Sampler > Add > Post Processors > JSR223 PostProcessor
The element is not added and an error is logged:
```
2021-03-17 09:21:03,282 ERROR o.a.j.JMeter: Uncaught exception in thread Thread[AWT-EventQueue-0,6,main]
java.lang.ExceptionInInitializerError: null
	at org.codehaus.groovy.jsr223.GroovyScriptEngineFactory.getLanguageVersion(GroovyScriptEngineFactory.java:95) ~[groovy-jsr223-3.0.7.jar:3.0.7]
	at org.apache.jmeter.util.JSR223BeanInfoSupport.<clinit>(JSR223BeanInfoSupport.java:69) ~[ApacheJMeter_core.jar:5.4.1]
	at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[?:?]
	at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:?]
	at java.lang.reflect.Constructor.newInstance(Constructor.java:490) ~[?:?]
	at java.lang.Class.newInstance(Class.java:584) ~[?:?]
	at com.sun.beans.finder.InstanceFinder.instantiate(InstanceFinder.java:98) ~[?:?]
	at com.sun.beans.finder.InstanceFinder.find(InstanceFinder.java:66) ~[?:?]
	at com.sun.beans.finder.BeanInfoFinder.find(BeanInfoFinder.java:41) ~[?:?]
	at java.beans.Introspector.findExplicitBeanInfo(Introspector.java:484) ~[?:?]
	at java.beans.Introspector.<init>(Introspector.java:434) ~[?:?]
	at java.beans.Introspector.getBeanInfo(Introspector.java:205) ~[?:?]
	at org.apache.jmeter.testbeans.gui.TestBeanGUI.<init>(TestBeanGUI.java:162) ~[ApacheJMeter_core.jar:5.4.1]
	at org.apache.jmeter.gui.GuiPackage.getGuiFromCache(GuiPackage.java:417) ~[ApacheJMeter_core.jar:5.4.1]
	at org.apache.jmeter.gui.GuiPackage.createTestElement(GuiPackage.java:368) ~[ApacheJMeter_core.jar:5.4.1]
	at org.apache.jmeter.gui.action.AddToTree.doAction(AddToTree.java:67) ~[ApacheJMeter_core.jar:5.4.1]
	at org.apache.jmeter.gui.action.ActionRouter.performAction(ActionRouter.java:87) ~[ApacheJMeter_core.jar:5.4.1]
	at org.apache.jmeter.gui.action.ActionRouter.lambda$actionPerformed$0(ActionRouter.java:69) ~[ApacheJMeter_core.jar:5.4.1]
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313) ~[?:?]
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:770) ~[?:?]
	at java.awt.EventQueue$4.run(EventQueue.java:721) ~[?:?]
	at java.awt.EventQueue$4.run(EventQueue.java:715) ~[?:?]
	at java.security.AccessController.doPrivileged(Native Method) ~[?:?]
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85) ~[?:?]
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:740) ~[?:?]
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203) [?:?]
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124) [?:?]
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113) [?:?]
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109) [?:?]
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101) [?:?]
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:90) [?:?]
Caused by: groovy.lang.GroovyRuntimeException: Conflicting module versions. Module [groovy-datetime is loaded in version 3.0.7 and you are trying to load version 3.0.3
	at org.codehaus.groovy.runtime.metaclass.MetaClassRegistryImpl$DefaultModuleListener.onModule(MetaClassRegistryImpl.java:510) ~[groovy-3.0.7.jar:3.0.7]
	at org.codehaus.groovy.runtime.m12n.ExtensionModuleScanner.scanExtensionModuleFromProperties(ExtensionModuleScanner.java:87) ~[groovy-3.0.7.jar:3.0.7]
	at org.codehaus.groovy.runtime.m12n.ExtensionModuleScanner.scanExtensionModuleFromMetaInf(ExtensionModuleScanner.java:81) ~[groovy-3.0.7.jar:3.0.7]
	at org.codehaus.groovy.runtime.m12n.ExtensionModuleScanner.scanClasspathModulesFrom(ExtensionModuleScanner.java:63) ~[groovy-3.0.7.jar:3.0.7]
	at org.codehaus.groovy.runtime.m12n.ExtensionModuleScanner.scanClasspathModules(ExtensionModuleScanner.java:54) ~[groovy-3.0.7.jar:3.0.7]
	at org.codehaus.groovy.runtime.metaclass.MetaClassRegistryImpl.<init>(MetaClassRegistryImpl.java:125) ~[groovy-3.0.7.jar:3.0.7]
	at org.codehaus.groovy.runtime.metaclass.MetaClassRegistryImpl.<init>(MetaClassRegistryImpl.java:86) ~[groovy-3.0.7.jar:3.0.7]
	at groovy.lang.GroovySystem.<clinit>(GroovySystem.java:37) ~[groovy-3.0.7.jar:3.0.7]
	... 32 more
```

I do not have any other plugins, and when I remove your plugin from /lib/ext folder I can add the "JSR223 PostProcessor" in JMeter GUI.

My guess is that the root issue is that you include libraries in your release jar which are delivered by JMeter out of the box.
And these version no longer match, depending on which version of JMeter one uses.
Is it possible to exclude libraries in the release jar which are already included by JMeter (e.g. Groovy, Groovy Datetime etc.)

Thanks in advance :)

**Comment by cah-jyoti-prakash** [comment]
What I found was that, if you run this plugin in Jmeter, it would work.
Alternatively you can use the source code and update the version and build new jar for Jmeter 5.4

**Comment by mderevyankoaqa** [comment]
@dsteindo, @cah-jyoti-prakash  guys are you really sure that issue with adding JSR223 PostProcessor has to do with plugin? 
I've faced with that issue in Jmeter GUI when I opened different Jmeter versions 5.4.1 and 5.3.0 for example and tried to copy something from 5.3.0 to 5.4.1. I suppose it's runtime issue, due to breaking changes.. You can try remove "jmeter-influxdb2-listener-plugin" and take a look at it again ..  Please let me know the result ..
I know that maybe plugin has more dependencies the needed.. need to optimize that later – has no time for now. Maybe we need to release the special version for the Jmeter 5.4.1 in case if its compatibility issue.

**Comment by dsteindo** [comment]
Hi @mderevyankoaqa , @cah-jyoti-prakash 
Thank you for the replies

To answer your questions:
- I have a clean installation of JMeter 5.4.1, freshly downloaded from https://jmeter.apache.org/download_jmeter.cgi
- There are no additional plugins
- They have a newer groovly library (3.0.7) out of the box in this version than in your 1.0 plugin release (3.0.3)
- As  mentioned in the initial comment, JSR223 Post Processor can be added without the "jmeter-influxdb2-listener-plugin" plugin

Regarding the new release:
- Replaced the "jmeter-plugin-influxdb2-listener-1.0.jar" with "jmeter-plugin-influxdb-listener-1.1.jar" in /lib/ext folder.
- Now JSR223 PostProcessor can be added
- Tested Grafana integration and it works with new 1.1 release version

However the issue might happen again in the future with newer JMeter versions.

For now it works, Thank you :)

**Comment by mderevyankoaqa** [comment]
@dsteindo very cool. Thank you so much! Thinking about how to improve .. I've updated the other libs versions as well, will push a new release today evening. 

BR, 
Mike

**Comment by mderevyankoaqa** [comment]
Fixed in [1.1 relase](https://github.com/mderevyankoaqa/jmeter-influxdb2-listener-plugin/releases/tag/v1.1)

**Comment by Goat2285** [comment]
Hey @mderevyankoaqa 

this issue once again popped its head with Jmeter 5.6.2
with you lib - cant add the jsre
without your lib, cant. same error
